packages/visx-text/src/hooks/useText.ts

```diff
@@ -0,0 +1,123 @@
+import { useMemo } from 'react';
+import reduceCSSCalc from 'reduce-css-calc';
+import { TextProps, WordsWithWidth } from '../types';
+import getStringWidth from '../util/getStringWidth';
+
+function isNumber(val: unknown): val is number {
+  return typeof val === 'number';
+}
+
+function isValidXOrY(xOrY: string | number | undefined) {
+  return (
+    // number that is not NaN or Infinity
+    (typeof xOrY === 'number' && Number.isFinite(xOrY)) ||
+    // for percentage
+    typeof xOrY === 'string'
+  );
+}
+
+export default function useText(
+  props: TextProps,
+): {
+  wordsByLines: WordsWithWidth[];
+  startDy: string;
+  transform: string;
+} {
+  const {
+    verticalAnchor = 'end',
+    scaleToFit = false,
+    angle,
+    width,
+    lineHeight = '1em',
+    capHeight = '0.71em', // Magic number from d3
+    children,
+    style,
+    ...textProps
+  } = props;
+
+  const { x = 0, y = 0 } = textProps;
+  const isXOrYValid = !isValidXOrY(x) || !isValidXOrY(y);
```

Comment:
naming nit – I think this actual should be `isXOrYInvalid` (invalid, not valid)

```suggestion
  const isXOrYInValid = !isValidXOrY(x) || !isValidXOrY(y);
```